### ISSUE TITLE
Policy on breaking changes

### ISSUE THREAD

**Issue by espen** [floere/phony#255]
The latest update brings a breaking change for formatted phone numbers.

Previously with 2.13.0 this would return a phone number without spaces and parentheses.
```Phony.formatted(phone_number, :format => :international, :spaces => '')```

now with version 2.14.0 I need to use this:
```Phony.formatted(phone_number, :format => :international, :spaces => '', :local_spaces => '', :parentheses => false)```

Simple to fix and I'm not a semver-fanatic, but would be nice to know if you have any thoughts of versioning and breaking changes?

**Comment by floere** [comment]
Hi @espen, can you provide an example, please?

**Comment by espen** [comment]
The example is above? Just wondering if you have any thoughts on implementing such changes either through a major or minor version, or if patch versions also may include breaking changes such as this example.

I've been using this gem for a while now and very happy with it :) Just want to clarify what to expect from new versions as this caught be my surprise and broke my tests.

**Comment by floere** [comment]
I'm trying to figure out why you think all numbers are generally working like that? Is this about the North American number plan?

**Comment by floere** [comment]
If so, after being asked too many times, and being referred to the NANP formatting including parentheses and dashes between local number parts, I finally gave up resisting when https://github.com/floere/phony/issues/252 came in.

But, to answer your question: a change in formatting rules is by Phony definition not breaking. A public interface change would be breaking. The idea here is that Phony users are assumed to trust Phony to gradually improve the formatting for all countries, for example like the NANP area, which got a big change with 2.14.0.

Consider my alternative: if ever I changed any country, and it would be a breaking change with a changing major version, phony would be at about major version 300.

This is why I decided Phony to not "break" when the formatting of a country changes, but just on interface changes.

To sum up, major version change: The public interface had a breaking change.
Minor version change: The public interface had a method added. A country had its formatting changed.
Patch: Everything else. For example, a country had its rules slightly changed.

So if you want to test against formatted numbers, then I am afraid you have to expect that with every release of Phony. I try not to have it happen too often, but it can happen.

I'm interested to hear your thoughts if you have any :)

**Comment by espen** [comment]
```
phone_number = Phony.normalize("+1 (541) 754-3010")
Phony.format(phone_number, :format => :international, :spaces => '')
```

Version 2.13.0 => "+15417543010"
Version 2.14.0 => "+1 (541) 754-3010"

**Comment by floere** [comment]
I cannot reproduce the behaviour with version 2.14.1:

```
ruby -e 'require "phony"; p Phony.format(Phony.normalize("+1 (541) 754-3010"), :format => :international, :spaces => "")'
```

Returns `"+1(541)754-3010"`.

**Comment by espen** [comment]
Thanks for the info. Understand what you say about country formatting rules and changes. In this case it seems to me more than just that though since the method is adding more arguments and this is causing initial behaviour to break: https://github.com/floere/phony/commit/1bd48bd4e5cfa859f6e3f13eaaf2a602f823594a

And regarding 2.14.1, well it still has parentheses which was introduced in that commit.

**Comment by floere** [comment]
@espen The commit you are referring to introduced the option to explicitly set the parentheses option to `false`. This is an added option to the API, which is why it increased the minor version number.

Based on feedback from #252, I changed the default of NANP (Canada, Mexico, US) to use parentheses around the NDC, and dashes in the local number. That necessitated that you can force parentheses to not show, hence the commit above. In itself, it is also a significant change to the NANP format, but since I need a breaking API change to increase the major number, it is only a minor change.

**Comment by floere** [comment]
In any case, I hope you understand that I understand that breaking tests can be annoying. Cheers! I'm closing this because I answered your question – just reopen if you have more questions.

**Comment by espen** [comment]
Ok, thanks for the info! :)
